pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -389,14 +391,29 @@ func (d *DeploymentController) render(templateName string, mi TemplateInput) ([]
 	if template == nil {
 		return nil, fmt.Errorf("no %q template defined", templateName)
 	}
+	proxyConfig := cfg.MeshConfig.GetDefaultConfig()
+	if d.env.PushContext != nil && d.env.PushContext.ProxyConfigs != nil {
+		if generatedProxyConfig := d.env.PushContext.ProxyConfigs.EffectiveProxyConfig(
+			&model.NodeMetadata{
+				Namespace: mi.Namespace,
+				// We need deployment labels here, so we have to check if the deployment exists and if so, get its labels.
+				// However, it won't exist on first creation of a gateway, so we might fall back
+				// to the default labels from the template: service.istio.io/canonical-name, istio.io/gateway-name.
```

Comment:
I don't understand this comment

Reply:
Ok, I will try to clarify. So the point is that to support ProxyConfig `spec.selector.matchLabels` the deployment controller must know the labels from the pod template in the gateway deployment, so before rendering template we would have to get deployment, but it will not exist during first rendering. So if ProxyConfig was created for a particular workload, e.g. `spec.selector.matchLabels: [istio.io/gateway-name: gateway]`, but a gateway does exist yet, the ProxyConfig won't be applied. Therefore we would have to check if ProxyConfig exists for default labels: service.istio.io/canonical-name, istio.io/gateway-name, because these labels will be set in the gateway pods.